File: src/main/java/cat/mvmike/numbertotext/magnitude/Cents.java
package cat.mvmike.numbertotext.magnitude;

import static cat.mvmike.numbertotext.language.Constants.DEC_CURRENCY;
import static cat.mvmike.numbertotext.language.Constants.DEC_SEPARATOR;
import static cat.mvmike.numbertotext.language.Constants.EMPTY;
import static cat.mvmike.numbertotext.language.Number.N_0;
import static cat.mvmike.numbertotext.language.Number.N_1;
import static cat.mvmike.numbertotext.language.Constants.PLURAL;
import static cat.mvmike.numbertotext.language.Constants.SPACE;
import static cat.mvmike.numbertotext.util.StringUtils.isEmpty;

public class Cents extends BaseMagnitude {

    public Cents(int number) {
        super(number);
    }

    @Override
    public String get() {
        String decimalPart = new Tens(number).get();
        return isEmpty(decimalPart) || N_0.getLiteral().equals(decimalPart) ?
                EMPTY : SPACE + DEC_SEPARATOR + SPACE + decimalPart;
    }

    @Override
    public String getCurrency(final String currency) {
        if (number == N_0.getValue() || isEmpty(currency))
            return EMPTY;

        return number == N_1.getValue() ? SPACE + DEC_CURRENCY : SPACE + DEC_CURRENCY + PLURAL;
    }
}

File: src/main/kotlin/cat/mvmike/numbertotext/magnitude/Cents.kt
// Copyright (c) 2016, Miquel Martí <miquelmarti111@gmail.com>
// See LICENSE for licensing information
package cat.mvmike.numbertotext.magnitude

import cat.mvmike.numbertotext.language.Constants.DEC_CURRENCY
import cat.mvmike.numbertotext.language.Constants.DEC_SEPARATOR
import cat.mvmike.numbertotext.language.Constants.EMPTY
import cat.mvmike.numbertotext.language.Constants.PLURAL
import cat.mvmike.numbertotext.language.Constants.SPACE
import cat.mvmike.numbertotext.language.NumberLiteral.N_0
import cat.mvmike.numbertotext.language.NumberLiteral.N_1
import cat.mvmike.numbertotext.language.isEqualTo

class Cents(
    private val number: Int
) {

    fun get(): String {
        val decimalPart = Tens(number).get()
        return when {
            decimalPart.isEmpty() -> EMPTY
            decimalPart.isEqualTo(N_0) -> EMPTY
            else -> SPACE + DEC_SEPARATOR + SPACE + decimalPart
        }
    }

    fun getCurrency(currency: String?) = when {
        currency.isNullOrEmpty() -> EMPTY
        number.isEqualTo(N_0) -> EMPTY
        number.isEqualTo(N_1) -> SPACE + DEC_CURRENCY
        else -> SPACE + DEC_CURRENCY + PLURAL
    }
}